### PR TITLE
Raise a warning for the use of short-form parameters when long-forms are available

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Address the comment in https://github.com/GenericMappingTools/pygmt/issues/1203#issuecomment-845350743.

When long-form parameters are available, we should recommend users not to use
the short-form parameters.

For example, running the following example:
```
import pygmt

fig = pygmt.Figure()
fig.basemap(region=[0, 10, 0, 10], B=True, J="X10c")
fig.savefig("map.pdf")
```
will raise warnings like:
```
test.py:4: FutureWarning: Short-form parameter (J) is not recommended. Use long-form parameter 'projection' instead.
  fig.basemap(region=[0, 10, 0, 10], B=True, J="X10c")
test.py:4: FutureWarning: Short-form parameter (B) is not recommended. Use long-form parameter 'frame' instead.
  fig.basemap(region=[0, 10, 0, 10], B=True, J="X10c")
```

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/helpers/decorators.py [hunk 1]
```diff
@@ -375,10 +375,16 @@ def new_module(*args, **kwargs):
...
+                        f"Parameters in short-form ({arg}) and long-form ({alias}) can't coexist."
                     )
                 if alias in kwargs:
                     kwargs[arg] = kwargs.pop(alias)
+                elif arg in kwargs:
+                    msg = (
+                        f"Short-form parameter ({arg}) is not recommended. "
+                        f"Use long-form parameter '{alias}' instead."
+                    )
+                    warnings.warn(msg, category=FutureWarning, stacklevel=2)
```
Comment (weiji14) [inline pygmt/helpers/decorators.py:387]: Using `FutureWarning` implies a deprecation at some future date. Do you want to either:

1. set a specific PyGMT version (e.g. v0.5.0) to deprecate this 'feature'
2. Raise a [SyntaxWarning](https://docs.python.org/3/library/exceptions.html#SyntaxWarning) instead
3. Raise a [SyntaxError](https://docs.python.org/3/library/exceptions.html#SyntaxError) or GMTInvalidInput error?

File: pygmt/helpers/decorators.py [hunk 2]
```diff
@@ -375,10 +375,16 @@ def new_module(*args, **kwargs):
             for arg, alias in aliases.items():
```
Comment (weiji14) [inline pygmt/helpers/decorators.py:375]: I find the variables 'arg' and 'alias' confusing. Maybe we should be more explicit and call them 'short_param' (e.g. -J) and 'long_alias' (e.g. projection) instead? What do you think?

```suggestion
            for short_param, long_alias in aliases.items():
```

File: pygmt/helpers/decorators.py (hunk 1 above)
Comment (seisman) [reply pygmt/helpers/decorators.py:387]: > Using `FutureWarning` implies a deprecation at some future date. Do you want to either:
> 
> 1. set a specific PyGMT version (e.g. v0.5.0) to deprecate this 'feature'

As mentioned in https://github.com/GenericMappingTools/pygmt/issues/1203#issuecomment-845253702, I don't think we will fully deprecate this feature before PyGMT v1.0.0.

> 2. Raise a [SyntaxWarning](https://docs.python.org/3/library/exceptions.html#SyntaxWarning) instead

Sounds good to me.

> 3. Raise a [SyntaxError](https://docs.python.org/3/library/exceptions.html#SyntaxError) or GMTInvalidInput error?

I'm afraid it's a little too aggressive. BTW, the "single-letter" parameter is also used a lot in the PyGMT tests. Raising any errors will bring us troubles.

Comment (seisman) [reply pygmt/helpers/decorators.py]: Changed to `SyntaxWarning` in commit 23e9dd67a999cb0c984dea13fb993434f4593944.

File: pygmt/helpers/decorators.py (hunk 2 above)
Comment (seisman) [reply pygmt/helpers/decorators.py]: Done in 30cf3e2cd2f7e33ad2fbc37cf081fdc02442899a.

Comment (weiji14) [reply pygmt/helpers/decorators.py]: Looks much better, thanks!

**Review by weiji14 — APPROVED** [review]
Ok to approve. Technically we should have written a test for this, but there are actually some tests still using short parameter names (as mentioned in https://github.com/GenericMappingTools/pygmt/pull/1316#discussion_r646221803) so I think it's alright to just merge it in as is.

**Pull request merged**